Added validation of incoming DTO requests

DOGGER152 opened the pull request.

DOGGER152 committed:
Added validation of incoming DTO requests

iv21dk commented on pom.xml:
```diff
@@ -29,6 +29,11 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
```
The dependency is not needed

iv21dk commented on pom.xml:
```diff
@@ -40,6 +45,16 @@
+            <artifactId>spring-boot-starter-validation</artifactId>
+        </dependency>
+
+        <dependency>
```
The dependency is not needed

iv21dk commented on src/main/java/com/springweb/dv_spring_web_mongo/dto/ProjectDTO.java:
```diff
@@ -14,9 +15,12 @@ public ProjectDTO(String id, String projectName) {
     }
 
+
+    @NotNull
```
Now I cannot create or update a project without specifying the `id` property of body. But in fact the field isn't used on backend side. The `id` field souldn't be required, or maybe it is better to intrioduce new dto object for correspond methods without `id` field?

iv21dk commented on src/main/java/com/springweb/dv_spring_web_mongo/controller/ProjectController.java:
```diff
@@ -25,12 +27,12 @@ public ProjectDTO findProjectById(@PathVariable String id) {
 
     @PostMapping()
-    public void addNewProject(@RequestBody ProjectDTO projectDTO) {
+    public void addNewProject(@Valid @RequestBody ProjectDTO projectDTO) {
```
It would be nice if server returned error code with data what is the set of incorrect or missing fields

DOGGER152 committed:
Unused  dependencies deleted

DOGGER152 committed:
Added Non-valid DTO exception handling.
Removed ID field from DTO.
Added lombok for DTO
Tests refactored for because of new changes

iv21dk commented on src/main/java/com/springweb/dv_spring_web_mongo/dto/ProjectDTO.java:
```diff
@@ -1,49 +1,22 @@
-    }
+import javax.validation.constraints.NotNull;
 
-    @Id
```
It is not good idea to remove `id` field from here because the dto object is used in GET /api/projects method and it is important to have this field in result of method
